docs/content/sips/016-component-imports.md

```diff
@@ -0,0 +1,53 @@
+title = "SIP 016 - Spin Component Imports" 
+template = "main" 
+date = "2023-09-22T16:27:43Z"
+---
+
+Summary: A proposal to introduce component imports to the Spin 2.0 manifest.
+
+Owner: <brian.hardock@fermyon.com>
+
+Created: Sep 19, 2023
+
+## Background / Goals
+The Spin 2.0 framework will support executing preview2 WebAssembly components. These components will include `import` and `export` statements. This manifest proposal is to enable Spin developers the ability to expressly configure how to satisfy the imports of their components. Furthermore, developers should be enabled to use the exports of a component to satisfy the imports of another.
+
+## Proposal
+
+### Changes to Spin Manifest 
+To support component imports, a new section will be added to the Spin 2.0 manifest design, `[component.<component-id>.import.<import-id>]`. The following sections describe the various options for specifying how to satisfy imports.
+
+> NOTE: `import-id` refers to one of the several optional for naming imports as per the [Component Model Explainer](https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md#import-and-export-definitions).
+
+### Reference by ID
+Imports can be satisfied by associating an import with a component ID which links to a component defined elsewhere in the manifest.
+
+```toml
+[component.foobar.import."foo"]
+source = "other"
+
+[component.other]
+source = { path = "bar.wasm" }
```

Comment:
What would happen if I specified `files` here?

Reply:
I was imagining using [wasi-virt](https://github.com/bytecodealliance/WASI-Virt); similarly for `environment`. I haven't experimented with this just yet so not entirely sure how viable this is.

Do you have any thoughts?

Reply:
If we don't go down the route of static composition then each component could have their own instance of wasi.

Reply:
> each component could have their own instance of wasi.

They can't currently; `wasmtime-wasi` (and bindgen in general?) doesn't support linking to anything other than root instances. That is something that could be fixed though.

Reply:
If you do dynamic linking, each instance can be backed by a completely separate store so yes it can have different wasi instances. This is how wepl works.

Reply:
Separate stores would mean we can't use resources, right?